main.go

```diff
@@ -0,0 +1,363 @@
+	}
+
+	//db.addressable.createIndex({name: 1}, {unique: true});
+	err = db.C("addressable").EnsureIndex(mgo.Index{Key: []string{"name"}, Name: "name_1", Unique: true})
+	if err != nil {
+		LoggingClient.Error("Error during execution: " + err.Error())
+	}
+
+	command := mgo.Collection{
+		Database: &db,
+		Name:     "command",
+		FullName: "db.command",
+	}
+	err = command.Create(&mgo.CollectionInfo{})
+	if err != nil {
+		LoggingClient.Error("Error during execution: " + err.Error())
+	}
+
+	//db.device.createIndex({name: 1}, {unique: true});
+	err = db.C("device").EnsureIndex(mgo.Index{Key: []string{"name"}, Name: "name_1", Unique: true})
+	if err != nil {
+		LoggingClient.Error("Error during execution: " + err.Error())
+	}
+
+	//db.deviceManager.createIndex({name: 1}, {unique: true});
+	err = db.C("deviceManager").EnsureIndex(mgo.Index{Key: []string{"name"}, Name: "name_1", Unique: true})
+	if err != nil {
+		LoggingClient.Error("Error during execution: " + err.Error())
+	}
+
+	//db.deviceProfile.createIndex({name: 1}, {unique: true});
+	err = db.C("deviceProfile").EnsureIndex(mgo.Index{Key: []string{"name"}, Name: "name_1", Unique: true})
+	if err != nil {
+		LoggingClient.Error("Error during execution: " + err.Error())
+	}
+
+	//db.deviceReport.createIndex({name: 1}, {unique: true});
+	err = db.C("deviceReport").EnsureIndex(mgo.Index{Key: []string{"name"}, Name: "name_1", Unique: true})
+	if err != nil {
+		LoggingClient.Error("Error during execution: " + err.Error())
+	}
+
+	//db.deviceService.createIndex({name: 1}, {unique: true});
+	err = db.C("deviceService").EnsureIndex(mgo.Index{Key: []string{"name"}, Name: "name_1", Unique: true})
+	if err != nil {
+		LoggingClient.Error("Error during execution: " + err.Error())
+	}
+
+	//db.provisionWatcher.createIndex({name: 1}, {unique: true});
+	err = db.C("provisionWatcher").EnsureIndex(mgo.Index{Key: []string{"name"}, Name: "name_1", Unique: true})
+	if err != nil {
+		LoggingClient.Error("Error during execution: " + err.Error())
+	}
+
+	//db.schedule.createIndex({name: 1}, {unique: true});
+	err = db.C("schedule").EnsureIndex(mgo.Index{Key: []string{"name"}, Name: "name_1", Unique: true})
+	if err != nil {
+		LoggingClient.Error("Error during execution: " + err.Error())
+	}
+
+	//db.scheduleEvent.createIndex({name: 1}, {unique: true});
+	err = db.C("scheduleEvent").EnsureIndex(mgo.Index{Key: []string{"name"}, Name: "name_1", Unique: true})
+	if err != nil {
+		LoggingClient.Error("Error during execution: " + err.Error())
+	}
+
+	db = mgo.Database{
+		Session: session,
+		Name:    "coredata",
+	}
+
+	err = db.UpsertUser(&mgo.User{
+		Username: "core",
+		Password: "password",
+		Roles: []mgo.Role{
+			mgo.RoleReadWrite,
+		},
+	})
+	if err != nil {
+		LoggingClient.Error("Error during execution: " + err.Error())
+	}
+
+	//db.event.createIndex({"device": 1}, {unique: false});
+	err = db.C("event").EnsureIndex(mgo.Index{Key: []string{"name"}, Name: "name_1", Unique: true})
```

Comment:
In original script, unique == false
https://github.com/edgexfoundry/docker-edgex-mongo/blob/b88dad2df2562f515b62e0cb486db88385442658/init_mongo.js#L83